lib/tapioca/compilers/dsl/active_record_typed_store.rb

```diff
@@ -0,0 +1,173 @@
+      #
+      # For example, with the following ActiveRecord class:
+      #
+      # ~~~rb
+      # # post.rb
+      # class Post < ApplicationRecord
+      #   typed_store :metadata do |s|
+      #     s.string(:reviewer, blank: false, accessor: false)
+      #     s.date(:review_date)
+      #     s.boolean(:reviewed, null: false, default: false)
+      #   end
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `post.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # post.rbi
+      # # typed: true
+      # class Post
+      #   sig { params(review_date: T.nilable(Date)).returns(T.nilable(Date)) }
+      #   def review_date=(review_date); end
+      #
+      #   sig { returns(T.nilable(Date)) }
+      #   def review_date; end
+      #
+      #   sig { returns(T.nilable(Date)) }
+      #   def review_date_was; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def review_date_changed?; end
+      #
+      #   sig { returns(T.nilable(Date)) }
+      #   def review_date_before_last_save; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def saved_change_to_review_date?; end
+      #
+      #   sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+      #   def review_date_change; end
+      #
+      #   sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+      #   def saved_change_to_review_date; end
+      #
+      #   sig { params(reviewd: T::Boolean).returns(T::Boolean) }
+      #   def reviewed=(reviewed); end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed_was; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed_changed?; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def reviewed_before_last_save; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def saved_change_to_reviewed?; end
+      #
+      #   sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+      #   def reviewed_change; end
+      #
+      #   sig { returns(T.nilable([T::Boolean, T::Boolean])) }
+      #   def saved_change_to_reviewed; end
+      # end
+      # ~~~
+      # end
+
+      class ActiveRecordTypedStore < Base
+        extend T::Sig
+
+        sig do
+          override
+            .params(
+              root: Parlour::RbiGenerator::Namespace,
+              constant: T.class_of(::ActiveRecord::Base)
+            )
+            .void
+        end
+        def decorate(root, constant)
+          stores = constant.typed_stores
+          return if stores.values.flat_map(&:accessors).empty?
+
+          root.path(constant) do |k|
+            stores.values.each do |store_data|
+              store_data.accessors.each do |accessor|
+                field = store_data.fields[accessor]
+                type = type_for(field.type_sym)
+                if type == "T.untyped"
+                  type = type.to_s
```

Comment:
Isn't `type` already a string here? 
```suggestion
                  type
```